### ISSUE TITLE
'dart run custom_lint' is not  working 

### ISSUE THREAD

**Issue by komalkhatri08** [invertase/dart_custom_lint#181]
**Describe the bug**
when I try to run dart run custom_lint in my application terminal (VS Code)  it says no issues found. where as the 'Problem' window shows the custom lint issues.

**To Reproduce**
pubspec.yaml of my application:- 
```
dev_dependencies:
  flutter_test:
    sdk: flutter
  json_annotation: ^4.8.1
  flutter_launcher_icons: "^0.13.1"
  lint: ^2.1.2
  custom_lint: ^0.5.0
  my_awesome_lints: 
    path: ../my_awesome_lints/
```

analysis_option.yaml file looks like :- 
```
analyzer: 
  plugins:
    custom_lint
```

![image](https://github.com/invertase/dart_custom_lint/assets/136062872/c1056132-23c6-4afb-9d57-2cd55855d341)
above image shows no issues found when i run 'dart run custom_lint'
![image](https://github.com/invertase/dart_custom_lint/assets/136062872/24b739c4-1cb6-42ec-865e-aaa8ac02e237)
above image shows custom lint i wrote is displayed in Problem window and custom_lint.log file .

**Expected behavior**
'dart run custom_lint' on terminal should list custom_lint issues I wrote. 
my main goal is to prevent commit using git hooks if 'flutter analyze' and custom lint issues exist in code.

**Comment by rrousselGit** [comment]
Is there a chance you could make a git repository to reproduce this?

**Comment by komalkhatri08** [comment]
Hi Remi,

Sorry I cannot do that.

Thanks,
Komal

**Comment by komalkhatri08** [comment]
@rrousselGit  any update?

**Comment by SunlightBro** [comment]
```yaml
analyzer: 
  plugins:
    custom_lint
```

is not valid YAML syntax, please check the [README](https://pub.dev/packages/custom_lint#using-our-custom-lint-package-in-an-application) or the official [YAML spec](https://yaml.org/spec/1.2.2/)

Instead it should be:
```yaml
analyzer: 
  plugins:
    - custom_lint
```

Also please for the future:
- Create a **minimal** reproducible code sample
  - remove **everything** else that is not needed to reproduce the issue
- Don't push it as a zip
- "**@.rrousselGit** any update?", in FOSS you should ask for help an not demand it.
  - Maybe thats just me (as a non-native english-speaker), that I interpret this as disrespectful.

This issue can be closed.

**Comment by komalkhatri08** [comment]
Hi, 
ok , First of all I really apologize if any one felt i demanded it. I really need help.  My intention was for help. 
Second, I kept a minimal code as possible i can.. I am new to uploading anything on github, so was not able to upload folders. Hence i Zipped it and uploaded. 
If you can help me with the procedure of uploading folders. Or uploading just the files should be fine ?

**Comment by rrousselGit** [comment]
When using git, you do not need to zip folders.
Git supports folders just fine :)

**Comment by komalkhatri08** [comment]
I tried uploading it ... it worked fine now , removed the .zip file. 
rrouseeIGit , i request you if you can look into this issue I am facing. 

1. I am using flutter project and created a package using 'dart create -t package my_lints' . 
2. I tried the same by creating dart demo app using 'dart create demo_app' command  and 'dart run custom_lint' worked fine. 
3. I guess then flutter framework comes in picture somehow it is not working. 

Would really appreciate if you can help in this scenario. 
Thanks

**Comment by rrousselGit** [comment]
Much better thanks!

But as @SunlightBro mentioned, your `analysis_options.yaml` has a mistake.

```diff
analyzer: 
  plugins:
-    custom_lint
+    - custom_lint
```

**Comment by komalkhatri08** [comment]
Yea I tried @SunlightBro suggestion, It applies lint rules on my code and gives me issues listed in 'Problems'. But when I use the command 'dart run custom_lint' command in terminal it still says no issues found

**Comment by rrousselGit** [comment]
Works fine for me:

```
remirousselet@Remis-MacBook-Pro custom_lint % dart run custom_lint
[avoid_plain_text_password] password
[amend_model_suffix] model.dart
  lib/main.dart:4:16 • Avoid using plain text passwords in code. • avoid_plain_text_password • INFO
  lib/model.dart:1:7 • Amend Model suffix in class names • amend_model_suffix • WARNING
  lib/services.dart:3:7 • Only one Service class allowed per file • one_service_class_per_file • INFO
```

**Comment by komalkhatri08** [comment]
ah.h.... worked fine. was a space issue in my analysis_options.yaml file. Thank you @rrousselGit @SunlightBro.
Also once again apologies @rrousselGit . 
One question still is open -> 
```
analyzer: 
  plugins:
    custom_lint
```

This line still list issues in the 'Problem' even though its not valid

**Comment by rrousselGit** [comment]
Well now that only a single plugin can be enabled at a time, maybe Dart supports single values now. Who knows

**Comment by komalkhatri08** [comment]
my main goal is to prevent commit using git hooks if 'flutter analyze' and custom lint issues exist in code. 
So when I write below code and try to commit the code It says cannot find custom_lint file in git bash. 
```
#!/bin/bash

# Run custom_lint
echo "Running custom_lint..."
dart run custom_lint


# Check the exit code of custom_lint
if [ $? -ne 0 ]; then
  echo "Custom linting issues found. Aborting commit."
  exit 1
fi

# Run flutter analyze
echo "Running flutter analyze..."
flutter analyze

# Check the exit code of flutter analyze
if [ $? -ne 0 ]; then
  echo "Linting issues found. Aborting commit."
  exit 1
fi
```
